### PR TITLE
Clean up frontend packages

### PR DESCRIPTION
**Pull request by nabramow**
We were getting a lot of package warnings, for missing peer dependencies. We also had many unused packages and missing packages.

I used `depcheck` to find any I missed: `yarn add depcheck` then `yarn depcheck` to see unused packages. Then delete it after you're done.

* I left the package `sharp`. When I remove that I get a bunch of next errors so decided not to poke the bear, might be something I'm missing even though that package doesn't seem to be imported anywhere.
* I left `jest-environment-jsdom` - the tests don't seem to run without it even though it's not referenced.
* `jest-junit` is used in a package.json script (it comes up as unused by depcheck).

**Web frontend checklist**
- [ x ] Formatted my code with `yarn format`
- [ x ] There are no warnings from `yarn lint --fix`
- [ x ] There are no console warnings when running the app
- [ x ] All tests pass
- [ x ] Clicked around my changes running locally and it works

### CODE REVIEW

File: app/web/.storybook/preview.js [hunk 1]
```diff
@@ -4,7 +4,6 @@ import {
   createTheme,
 } from "@mui/material/styles";
 import { QueryClient, QueryClientProvider } from "react-query";
-import { ThemeProvider as Emotion10ThemeProvider } from "emotion-theming";
```
Comment (nabramow) [inline app/web/.storybook/preview.js:7]: I added this while I was upgrading MUI and never used it. Forgot to remove it then.

File: app/web/features/auth/useAuthStore.test.ts [hunk 2]
```diff
@@ -1,8 +1,7 @@
-import { renderHook } from "@testing-library/react-hooks";
+import { act, renderHook } from "@testing-library/react-hooks";
 import { Empty } from "google-protobuf/google/protobuf/empty_pb";
 import { StatusCode } from "grpc-web";
 import { usePersistedState } from "platform/usePersistedState";
-import { act } from "react-test-renderer";
```
Comment (nabramow) [inline app/web/features/auth/useAuthStore.test.ts:5]: This was the only place we were using `react-test-renderer` so I switched to use it from `testing-library` instead so we can remove that package.

File: app/web/package.json [hunk 3]
```diff
@@ -22,41 +22,41 @@
     "typecheck": "tsc"
   },
   "dependencies": {
-    "@date-io/dayjs": "1.x",
```
Comment (nabramow) [inline app/web/package.json:25]: I forgot to remove this when I updated the datepicker.

File: app/web/package.json [hunk 4]
```diff
@@ -22,41 +22,41 @@
...
-    "@emotion/cache": "^11.13.1",
     "@emotion/react": "^11.13.5",
-    "@emotion/server": "^11.11.0",
```
Comment (nabramow) [inline app/web/package.json:28]: Was an experiment when upgrading MUI that I forgot to remove.

File: app/web/package.json [hunk 5]
```diff
@@ -22,41 +22,41 @@
...
     "@emotion/styled": "^11.13.5",
     "@fontsource/mansalva": "^5.1.0",
     "@fontsource/ubuntu": "^5.1.0",
     "@mui/icons-material": "^5.16.8",
     "@mui/lab": "^5.0.0-alpha.164",
     "@mui/material": "^5.16.7",
     "@mui/styles": "^5.16.7",
+    "@mui/system": "^5.4.1",
```
Comment (nabramow) [inline app/web/package.json:33]: unmet peer dependency for mui-x-datepickers

File: app/web/package.json [hunk 6]
```diff
@@ -22,41 +22,41 @@
...
     "@mui/x-date-pickers": "^5.0.20",
     "@sentry/nextjs": "^6.19.7",
     "@sentry/react": "^8.41.0",
     "@stripe/stripe-js": "^5.2.0",
     "@toast-ui/editor": "^3.2.2",
-    "@types/react-simple-maps": "^3.0.6",
     "classnames": "^2.5.1",
     "dayjs": "^1.11.13",
     "frontmatter-markdown-loader": "^3.7.0",
+    "geojson": "^0.5.0",
```
Comment (nabramow) [inline app/web/package.json:43]: We're importing this in app/web/features/search/MapWrapper.tsx but never installed the package so it was grabbing it from the node_modules instead, which isn't very reliable.

File: app/web/package.json [hunk 7]
```diff
@@ -22,41 +22,41 @@
...
     "@mui/material": "^5.16.7",
     "@mui/styles": "^5.16.7",
+    "@mui/system": "^5.4.1",
     "@mui/utils": "^5.16.6",
     "@mui/x-date-pickers": "^5.0.20",
     "@sentry/nextjs": "^6.19.7",
     "@sentry/react": "^8.41.0",
     "@stripe/stripe-js": "^5.2.0",
     "@toast-ui/editor": "^3.2.2",
-    "@types/react-simple-maps": "^3.0.6",
```
Comment (nabramow) [inline app/web/package.json:42]: Moved to devDependencies

File: app/web/package.json [hunk 8]
```diff
@@ -22,41 +22,41 @@
...
     "classnames": "^2.5.1",
     "dayjs": "^1.11.13",
     "frontmatter-markdown-loader": "^3.7.0",
+    "geojson": "^0.5.0",
     "glob": "^8.0.3",
     "google-protobuf": "^3.21.4",
     "grpc-web": "1.5.0",
+    "i18next": "^21.5.3",
```
Comment (nabramow) [inline app/web/package.json:47]: We're importing it in app/web/.storybook/i18n.ts but never installed the package.

File: app/web/package.json [hunk 9]
```diff
@@ -22,41 +22,41 @@
...
     "intersection-observer": "^0.12.2",
     "luhn": "^2.4.1",
     "maplibre-gl": "^4.7.1",
     "markdown-it": "^14.1.0",
     "next": "12.1.6",
     "next-i18next": "git+https://github.com/Couchers-org/next-i18next.git#39a50ec9",
+    "prop-types": "^15.7.2",
```
Comment (nabramow) [inline app/web/package.json:54]: unmet peer dependency for `react-simple-maps`

File: app/web/package.json [hunk 10]
```diff
@@ -22,41 +22,41 @@
...
     "react": "17.0.2",
     "react-dom": "17.0.2",
-    "react-error-boundary": "^4.1.2",
```
Comment (nabramow) [inline app/web/package.json:57]: unused

File: app/web/package.json [hunk 11]
```diff
@@ -22,41 +22,41 @@
...
     "react-gtm-module": "^2.0.11",
     "react-hook-form": "^6.15.8",
+    "react-i18next": "^11.8.13",
```
Comment (nabramow) [inline app/web/package.json:59]: We're importing this all over the place but never installed the package.

File: app/web/package.json [hunk 12]
```diff
@@ -71,25 +71,23 @@
     "@storybook/addon-essentials": "^6.5.10",
     "@storybook/addon-links": "^6.5.10",
     "@storybook/builder-webpack5": "^6.5.10",
-    "@storybook/cli": "^8.4.5",
```
Comment (nabramow) [inline app/web/package.json:74]: unused

File: app/web/package.json [hunk 13]
```diff
@@ -71,25 +71,23 @@
...
     "@storybook/manager-webpack5": "^6.5.10",
     "@storybook/react": "^6.5.10",
+    "@testing-library/dom": "7.21.4",
```
Comment (nabramow) [inline app/web/package.json:76]: unmet peer dependency for `testing-library/user-event`

File: app/web/package.json [hunk 14]
```diff
@@ -71,25 +71,23 @@
...
     "@testing-library/jest-dom": "^5.16.4",
     "@testing-library/react": "^12.1.5",
     "@testing-library/react-hooks": "^8.0.1",
     "@testing-library/user-event": "^13.4.1",
     "@types/css-mediaquery": "^0.1.4",
     "@types/geojson": "^7946.0.10",
     "@types/google-protobuf": "^3.15.12",
-    "@types/jest": "^28.1.6",
```
Comment (nabramow) [inline app/web/package.json:84]: unused

File: app/web/package.json [hunk 15]
```diff
@@ -71,25 +71,23 @@
...
     "@types/markdown-it": "^14.1.2",
     "@types/node": "^22.10.1",
     "@types/react": "^17.0.43",
-    "@types/react-dom": "^17.0.13",
```
Comment (nabramow) [inline app/web/package.json:88]: unused

File: app/web/package.json [hunk 16]
```diff
@@ -71,25 +71,23 @@
...
     "@types/react-gtm-module": "^2.0.4",
-    "@types/react-test-renderer": "^17.0.1",
```
Comment (nabramow) [inline app/web/package.json:90]: removing since the package was removed

File: app/web/package.json [hunk 17]
```diff
@@ -71,25 +71,23 @@
...
+    "@types/react-simple-maps": "^3.0.6",
     "@typescript-eslint/eslint-plugin": "^5.16.0",
-    "babel-loader": "^9.2.1",
```
Comment (nabramow) [inline app/web/package.json:92]: unused, it's not listed as the loader in the babel.config

File: app/web/package.json [hunk 18]
```diff
@@ -71,25 +71,23 @@
...
+    "@typescript-eslint/parser": "^5.0.0",
```
Comment (nabramow) [inline app/web/package.json:92]: unmet peer dependency for `@typescript-eslint/eslint-plugin`

File: app/web/package.json [hunk 19]
```diff
@@ -101,14 +99,14 @@
     "i18next-resources-to-backend": "^1.0.0",
     "jest": "^28.1.3",
     "jest-environment-jsdom": "^29.7.0",
-    "jest-junit": "^14.0.1",
```
Comment (nabramow) [inline app/web/package.json:104]: unused

File: app/web/package.json [hunk 20]
```diff
@@ -101,14 +99,14 @@
...
     "msw": "^1.2.1",
     "next-router-mock": "^0.7.4",
     "prettier": "^2.5.1",
-    "react-test-renderer": "^17.0.2",
+    "require-from-string": "^2.0.2",
```
Comment (nabramow) [inline app/web/package.json:108]: unmet peer dependency for `@storybook/react`

File: app/web/routes.ts [hunk 21]
```diff
@@ -5,7 +5,6 @@ Please make sure this file stays in sync with that file as well as
 //app/backend/src/couchers/urls.py
 */
 import { ReferenceType } from "proto/references_pb";
-import { ReferenceTypeStrings } from "service/references";
```
Comment (nabramow) [inline app/web/routes.ts:8]: never used

**Review by aapeliv — APPROVED** [review]
Awesome! Thank you for this PR. I was always wondering if those warnings were just expected or if there was some dep fixing that we were ignoring :see_no_evil:

File: app/web/package.json (hunk 15 above)
Comment (darrenvong) [reply app/web/package.json:88]: Hmm wouldn't this cause an error if we ever try to use anything from the `react-dom` package (e.g. createPortal) which is an important package for React web apps?

Comment (darrenvong) [reply app/web/package.json:88]: oh wait looks like the diff I commented on is outdated? I can still see it in the final diff 🤔

Comment (nabramow) [reply app/web/package.json:88]: Oh yeah whoops I put it back, I thought it got rid of the comment, sorry! Same with the jest types if you see that.

File: app/web/.storybook/preview.js (hunk 1 above)
Comment (darrenvong) [reply app/web/.storybook/preview.js:7]: I did wonder what that was but also forgot to ask in the original PR

File: app/web/package.json (hunk 4 above)
Comment (darrenvong) [reply app/web/package.json:28]: Interesting, it "sounds" like a package that'd be needed for server-side rendering but I trust that if things work, better to go without until we need it!

File: app/web/package.json [hunk 22]
```diff
@@ -22,41 +22,41 @@
...
     "glob": "^8.0.3",
     "google-protobuf": "^3.21.4",
     "grpc-web": "1.5.0",
+    "i18next": "^21.5.3",
     "intersection-observer": "^0.12.2",
     "luhn": "^2.4.1",
     "maplibre-gl": "^4.7.1",
     "markdown-it": "^14.1.0",
     "next": "12.1.6",
     "next-i18next": "git+https://github.com/Couchers-org/next-i18next.git#39a50ec9",
```
Comment (darrenvong) [inline app/web/package.json:53]: Probably one for a separate task... but I reckon we can move off this forked version of `next-i18next` so we are receiving updates again. I believe it was pinned to this as Lucas (one of the former active maintainers alongside with me!) made a fix upstream and it was unclear when they were merging and making a proper release with the fix. I believe the fix has been accepted now though so probably safe to go back to an official version instead.

Comment (darrenvong) [reply app/web/package.json:53]: Can confirm it was taken into upstream here! https://github.com/i18next/next-i18next/pull/1706

**Review by darrenvong — APPROVED** [review]
Yeah all looks good to me too! Thanks for tidying up those warnings 🌟

File: app/web/package.json (hunk 4 above)
Comment (nabramow) [reply app/web/package.json:28]: @darrenvong we will need it for the next version of MUI but need to update to Next.js v13 and above first.

File: app/web/package.json (hunk 22 above)
Comment (nabramow) [reply app/web/package.json:53]: Ah amazing I actually recently tagged Lucas in the PR he did this in asking the story behind it but even better that you know! That was on my radar to deal with eventually so I'm happy someone remembers why it was done so I don't have to stab in the dark!

**Pull request merged**